src/core/auth/qgsauthmanager.cpp

```diff
@@ -127,7 +127,7 @@ QSqlDatabase QgsAuthManager::authDatabaseConnection() const
     if ( QThread::currentThread() != QgsApplication::instance()->thread() )
     {
       QgsDebugMsgLevel( QStringLiteral( "Scheduled auth db remove on thread close" ), 0 );
-      connect( QThread::currentThread(), &QThread::finished, QThread::currentThread(), [connectionName]
+      connect( QThread::currentThread(), &QThread::finished, this, [connectionName]
```

Comment:
hmmmm... using "this" link the connection to the lifecicle of the authmanager (that is a singleton)... leaving QThread::currentThread() should give a more secure control of disconnect. Or you want to be sure that the lambda is called also when the thread is (eventually) removed before the lambda call?

Reply:
With the thread as the context this slot was never getting called. My understanding is that the connection will get destroyed alongside the sender (the thread) anyway.

Reply:
can be affected by the fact that before wasn't used the Qt::DirectConnection and the signal queued => thread destropyed and then labda never called?

Reply:
Ok, I've added a commit doing this. Preliminary testing looks good.